### ISSUE TITLE
More basic statistics

### ISSUE THREAD

**Issue by AaronEpiney** [idaholab/raven#279]
--------
Issue Description
--------
##### What did you expect to see happen?
It would be nice to have the possibility to compute basic statistics over time in addition to the time dependent ones that are already available.

For example, the available 'time dependent' statistics will compute the asked value (mean, variance, etc.) for every time-step of the input time histories and then output a time history for that value. The proposed new statistics should compute the asked value (mean, variance, etc) over time (instead of every time-step) and output only one value for the all the input time histories.

##### What did you see instead?
This is a proposition for a new capability in RAVEN.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [ ] 1. Is it tagged with a type: defect or improvement?
- [ ] 2. Is it tagged with a priority: critical, normal or minor?
- [ ] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [ ] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [ ] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [ ] 1. If the issue is a defect, is the defect fixed?
- [ ] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [ ] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [ ] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)?
- [ ] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided?

**Comment by PaulTalbot-INL** [comment]
For the record, a similar result can be obtained by using a History Set to Point Set interfaced postprocessor to turn a history into a set of time-based samples, then running basic statistics on this new point set.

One ambiguity to me is how to treat both "time" and "sample" at the same time.  It appears the proposed solution is to treat every time step in all samples as independent samples, which probably addresses the needs of most situations.

**Comment by alfoa** [comment]
Yes you are right. I thought about this solution but it becomes impractical when handling very large data set since it causes to double the memory (HS + PS (of same size)).
